File: cordapp-cpk2/src/test/resources/cordapp-import-package/src/main/java/com/example/importpackage/ExampleContract.java
package com.example.importpackage;

import net.corda.v5.ledger.contracts.Contract;
import net.corda.v5.ledger.transactions.LedgerTransaction;
import org.slf4j.LoggerFactory;
import org.slf4j.Logger;

public class ExampleContract implements Contract {
    private final Logger logger = LoggerFactory.getLogger(ExampleContract.class);

    @Override
    public void verify(LedgerTransaction ltx) {
        logger.info("Verify transaction: {}", ltx);
    }
}

File: cordapp-cpk2/src/test/resources/with-cordapp-name/contracts/src/main/kotlin/com/example/name/contracts/ExampleContract.kt
@file:Suppress("unused", "PackageDirectoryMismatch")
package com.example.name.contracts

import net.corda.v5.application.identity.AbstractParty
import net.corda.v5.ledger.contracts.Contract
import net.corda.v5.ledger.contracts.ContractState
import net.corda.v5.ledger.transactions.LedgerTransaction
import java.util.Collections.unmodifiableList

class ExampleContract : Contract {
    override fun verify(ltx: LedgerTransaction) {
        @Suppress("ObjectLiteralToLambda")
        val innerContract = object : Contract {
            override fun verify(ltx: LedgerTransaction) {
                println(ltx.toString())
            }
        }
        innerContract.verify(ltx)

        val lambdaContract = Contract { l -> println(l.toString()) }
        lambdaContract.verify(ltx)
    }

    class NestedContract : Contract {
        override fun verify(ltx: LedgerTransaction) = throw UnsupportedOperationException(this::class.java.name)
    }

    inner class InnerContract : Contract {
        override fun verify(ltx: LedgerTransaction) = throw UnsupportedOperationException(this::class.java.name)
    }
}

class ExampleState(val issuer: AbstractParty) : ContractState {
    override fun getParticipants(): List<AbstractParty> {
        return unmodifiableList(listOf(issuer))
    }
}
